test/jdk/java/net/httpclient/UserAuthWithAuthenticator.java

```diff
@@ -0,0 +1,375 @@
+                .setHeader("User-Agent", "myUserAgent")
+                .build();
+
+            var response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            assertEquals(200, response.statusCode());
+            assertEquals(data1, response.body());
+
+            var serverStr = serverMock.getRequest(1);
+            assertContains(serverStr, "/some_serv_url");
+            assertPattern(".*^User-Agent:.*myUserAgent.*", serverStr);
+            assertPattern(".*^Authorization:.*Basic " + encoded + ".*", serverStr);
+            System.out.println("testServerOnlyAuthenticator: OK");
+        } finally {
+            serverMock.stopMocker();
+        }
+    }
+
+    static void close(Closeable... clarray) {
+        for (Closeable c : clarray) {
+            try {
+                c.close();
+            } catch (Exception e) {}
+        }
+    }
+
+    static class Mocker extends Thread {
+        final ServerSocket ss;
+        final String[] responses;
+        volatile List<String> requests;
+        volatile InputStream in;
+        volatile OutputStream out;
+        volatile Socket s = null;
+
+        public Mocker(String[] responses) throws IOException {
+            this.ss = new ServerSocket(0);
+            this.responses = responses;
+            this.requests = new LinkedList<>();
+        }
+
+        public void stopMocker() {
+            close(ss, s, in, out);
+        }
+
+        public int getPort() {
+            return ss.getLocalPort();
+        }
+
+        public String baseURL() {
+            try {
+                return URIBuilder.newBuilder()
+                    .scheme("http")
+                    .loopback()
+                    .port(getPort())
+                    .build()
+                    .toString();
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        private String readRequest() throws IOException {
+            String req = "";
+            while (!req.endsWith("\r\n\r\n")) {
+                int x = in.read();
+                if (x == -1) {
+                    s.close();
+                    s = ss.accept();
+                    in = s.getInputStream();
+                    out = s.getOutputStream();
+                }
+                req += (char)x;
+            }
+            return req;
+        }
+
+        public String getRequest(int i) {
+            return requests.get(i);
+        }
+
+        public void run() {
+            try {
+                int index=0;
+                s = ss.accept();
+                in = s.getInputStream();
+                out = s.getOutputStream();
+                while (index < responses.length) {
+                    requests.add(readRequest());
+                    out.write(responses[index++].getBytes(US_ASCII));
+                }
+            } catch (Exception e) {
+                //e.printStackTrace();
+            }
```

Comment:
I think it might be OK to uncomment this and actually print the stacktrace, to help debug any unexpected failures.